gio/gappinfo.c

```diff
@@ -824,12 +846,68 @@ launch_default_with_portal_async (const char          *uri,
   g_free (real_uri);
 }
 
+static void
+launch_default_with_portal_sync (const char          *uri,
+                                 GAppLaunchContext   *context)
+{
+  GDBusConnection *session_bus;
+  GVariantBuilder opt_builder;
+  GVariant *res = NULL;
+  const char *parent_window = NULL;
+  char *real_uri;
+  GError *error = NULL;
+
+  session_bus = g_bus_get_sync (G_BUS_TYPE_SESSION, NULL, &error);
+  if (session_bus == NULL)
+    {
+      g_task_report_error (context, NULL, NULL, NULL, error);
+      return;
+    }
+
+  if (context && context->priv->envp)
```

Comment:
I usually also prefer to check variables directly but since in other places we're checking directly for NULL, we should be consistent here and do _context != NULL_.